policies.go

```diff
@@ -1088,6 +1127,48 @@ func (d *rackAwareRR) Reset()                              {}
 func (d *rackAwareRR) KeyspaceChanged(KeyspaceUpdateEvent) {}
 func (d *rackAwareRR) SetPartitioner(p string)             {}
 
+func (d *rackAwareRR) IsOperational(session *Session) error {
+	if !session.cfg.disableInit && !session.cfg.disableControlConn {
+		var datacenters []string
+		var racks []string
+		hosts, _, err := session.hostSource.GetHosts()
+		if err != nil {
+			return fmt.Errorf("gocql: unable to check if session is operational: %v", err)
+		}
+		for _, host := range hosts {
+			if !session.cfg.filterHost(host) {
+				datacenters = append(datacenters, host.DataCenter())
+				racks = append(racks, host.Rack())
+			}
+		}
+
+		foundDC := false
+		for _, dc := range datacenters {
+			if dc == d.localDC {
+				foundDC = true
+				break
+			}
+		}
+
+		foundRack := false
+		for _, rack := range racks {
+			if rack == d.localRack {
+				foundRack = true
+				break
+			}
+		}
+
+		if !foundDC {
+			return fmt.Errorf("gocql: datacenter provided in the policy was not found in the topology")
+		}
+		if !foundRack {
+			return fmt.Errorf("gocql: rack provided in the policy was not found in the topology")
+		}
+	}
+
+	return nil
+}
```

Comment:
```suggestion
	if session.cfg.disableInit || session.cfg.disableControlConn {
		return nil
	}
	hosts, _, err := session.hostSource.GetHosts()
	if err != nil {
		return fmt.Errorf("gocql: unable to check if session is operational: %v", err)
	}
	for _, host := range hosts {
		if !session.cfg.filterHost(host) && host.DataCenter() == d.localDC && host.Rack() == d.localRack {
			// Policy can work properly only if there is at least one host from target DC+Rack
			// No need to check host status, since it could be down due to the outage
			// We only need to make sure that policy is not misconfigured with wrong DC+Rack
			return nil
		}
	}

	return fmt.Errorf("gocql: rack %s/%s was not found in the topology", d.localDC, d.localRack)
}
```

Reply:
done